src/main/java/com/flightstats/hub/dao/aws/S3LargeContentDao.java

```diff
@@ -184,26 +182,26 @@ public Content get(final String channelName, final ContentKey key) {
         try {
             return getS3Object(channelName, key);
         } catch (SocketTimeoutException e) {
-            logger.warn("SocketTimeoutException : unable to read " + channelName + " " + key);
+            log.warn("SocketTimeoutException : unable to read " + channelName + " " + key);
             try {
                 return getS3Object(channelName, key);
             } catch (Exception e2) {
-                logger.warn("unable to read second time " + channelName + " " + key + " " + e.getMessage(), e2);
+                log.warn("unable to read second time " + channelName + " " + key + " " + e.getMessage(), e2);
                 return null;
             }
         } catch (Exception e) {
-            logger.warn("unable to read " + channelName + " " + key, e);
+            log.warn("unable to read " + channelName + " " + key, e);
             return null;
         } finally {
             ActiveTraces.getLocal().add("S3LargeContentDao.read completed");
         }
     }
 
     private Content getS3Object(String channelName, ContentKey key) throws IOException {
+
         long start = System.currentTimeMillis();
-        try {
-            GetObjectRequest request = new GetObjectRequest(s3BucketName.getS3BucketName(), getS3ContentKey(channelName, key, false));
-            S3Object object = s3Client.getObject(request);
+        GetObjectRequest request = new GetObjectRequest(s3BucketName.getS3BucketName(), getS3ContentKey(channelName, key, false));
+        try (S3Object object = s3Client.getObject(request)) {
```

Comment:
I'm not entirely sure this is going to do what you expect it to. According to the docs (https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/S3Object.html#getObjectContent-- and https://github.com/aws/aws-sdk-java/blob/master/src/samples/AmazonS3/S3Sample.java), the connection will remain open until the stream (`getObjectContent`) is read to completion or explicitly closed. Since this method just copies the stream into a `Content` object, I think we'll need to make sure all the callers that read the stream have a finally block that either calls `Content#getObjectContent().close()` or `Content#close()`. If possible, I'd write some tests to confirm.

Reply:
Yes correct. The integration tests passed locally on single instance. These tests failing in cluster mode.